IHP/NameSupport.hs

```diff
@@ -57,6 +61,31 @@ tableNameToControllerName tableName = do
         else ucfirst tableName
 {-# INLINABLE tableNameToControllerName #-}
 
+-- | Transforms a enum value to a name for a model
+--
+-- >>> enumValueToControllerName "happy"
+-- "Happy"
+--
+-- >>> enumValueToControllerName "very happy"
+-- "VeryHappy"
+--
+-- >>> enumValueToControllerName "very_happy"
+-- "VeryHappy"
+enumValueToControllerName :: Text -> Text
+enumValueToControllerName enumValue =
+    let
+        words :: [Inflector.SomeWord]
+        words = 
+                enumValue
+                |> splitOn " "
+                |> List.map (Inflector.parseSnakeCase [])
+                |> List.map (\case
+                        Left failed -> error (cs $ "enumValueToControllerName failed for " <> show failed)
+                        Right result -> result)
+                |> join
+    in
+        Inflector.camelizeCustom True words
```

Comment:
This PR results in camel-case data constructors for enums that are all-caps. As an example, an enum of `'USD'` is now produces a `Usd`, whereas the generated data constructor was previously `USD`. 

Is this the intended behaviour?

Reply:
Yes, this is expected behavior 👍